test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -0,0 +1,345 @@
+		await expect(
+			page.getByRole( 'document', {
+				name: 'Block: Heading',
+			} )
+		).toBeFocused();
+	} );
+
+	// Check for regression of https://github.com/WordPress/gutenberg/issues/39026.
+	test( 'selects the next block after removing the very first block', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Select the image block in List View.
+		await pageUtils.pressKeyTimes( 'ArrowUp', 2 );
+		await expect(
+			page
+				.getByRole( 'gridcell', {
+					name: 'Image link',
+				} )
+				.locator( 'a' )
+		).toBeFocused();
+		await page.keyboard.press( 'Enter' );
+
+		// Remove the Image block via its options menu in List View.
+		await page
+			.getByRole( 'button', { name: 'Options for Image block' } )
+			.click();
+		await page.getByRole( 'menuitem', { name: /Remove Image/i } ).click();
+
+		// Heading block should be selected as previous block.
+		await expect(
+			page.getByRole( 'document', {
+				name: 'Block: Heading',
+			} )
+		).toBeFocused();
+	} );
+
+	/**
+	 * When all the blocks gets removed from the editor, it inserts a default
+	 * paragraph block; make sure that paragraph block gets selected after
+	 * removing blocks from ListView.
+	 */
+	test( 'selects the default paragraph block after removing all blocks', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Select the Image block as well.
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowUp' );
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Image link',
+				selected: true,
+			} )
+			.waitFor();
+
+		// Remove both blocks.
+		await page
+			.getByRole( 'button', { name: 'Options for Image block' } )
+			.click();
+		await page.getByRole( 'menuitem', { name: /Remove blocks/i } ).click();
+
+		// Newly created paragraph block should be selected.
+		await expect(
+			page.getByRole( 'document', { name: /Empty block/i } )
+		).toBeFocused();
+	} );
+
+	test( 'expands nested list items', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Click first color option from the block placeholder's color picker to
+		// make the inner blocks appear.
+		await page
+			.getByRole( 'button', { name: /Color: /i } )
+			.first()
+			.click();
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Things start off expanded.
+		await page
+			.getByRole( 'gridcell', {
+				name: 'Cover link',
+				expanded: true,
+			} )
+			.waitFor();
```

Comment:
Ditto as what @Mamaduka suggested in https://github.com/WordPress/gutenberg/pull/47919/files#r1102633437.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/47919/commits/2193b792b6ea60076682b8b34a292f7783181c4c.